msal4j-sdk/src/integrationtest/java/com.microsoft.aad.msal4j/AcquireTokenSilentIT.java

```diff
@@ -344,6 +344,48 @@ void acquireTokenSilent_emptyScopeSet(String environment) throws Exception {
         assertEquals(result.accessToken(), silentResult.accessToken());
     }
 
+    @Test
+    public void acquireTokenSilent_ClaimsForceRefresh() throws Exception {
+        cfg = new Config(AzureEnvironment.AZURE);
+        User user = labUserProvider.getDefaultUser(AzureEnvironment.AZURE);
+
+        Set<String> scopes = new HashSet<>();
+        PublicClientApplication pca = PublicClientApplication.builder(
```

Comment:
I think unit tests should cover this not integration tests. Integration tests are expensive. There is no integration point that is worth testing unless you pass in some well-known claims and can assert the result. 

Unit tests should cover: 

- cca.AcquireTokenSilent 
- cca.AcquireTokenForClient
- pca.AcquireTokenSilent

Reply:
In the latest commit I added extra tests using some dummy cache data, to show that a token is returned when using the AcquireTokenSilent API without claims and not returned when claims are added to the request.

I did not add a specific test for the implicit silent call in the 'cca.AcquireTokenForClient' scenario, because internally it reaches the exact same code as the explicit AcquireTokenSilent call and that behavior is covered by other tests.